### PR TITLE
feat: Handle REGAPIC mix-ins

### PR DESCRIPTION
**Pull request by jskeet**
This requires two overall changes, both to PackageApiMetadata

- Include mix-in file descriptors in the list of descriptors for RestGrpcAdapter to use
- Expose HttpRule overrides from the service config (for mix-ins)

The latter requires https://github.com/googleapis/gax-dotnet/pull/616 to be merged, released, and then the project file updated (hence why this PR fails to build right now).

### CODE REVIEW

**Comment by jskeet** [comment]
If everyone is happy with the generated code, this will be ready to merge once there's a GAX alpha release that we can update to.

File: Google.Api.Generator.Tests/ProtoTests/Showcase/Google.Showcase.V1Beta1/IdentityClient.g.cs [hunk 1]
```diff
@@ -116,6 +120,16 @@ private IdentitySettings(IdentitySettings existing) : base(existing)
         /// </remarks>
         public gaxgrpc::CallSettings ListUsersSettings { get; set; } = gaxgrpc::CallSettings.FromExpiration(gax::Expiration.None);
 
+        /// <summary>
+        /// The settings to use for the <see cref="gcl::LocationsClient"/> associated with the client.
```
Comment (amanda-tarafa) [inline Google.Api.Generator.Tests/ProtoTests/Showcase/Google.Showcase.V1Beta1/IdentityClient.g.cs:124]: nit, clearer and for consistency with some other similar comments.
```suggestion
        /// The settings to use for the <see cref="gcl::LocationsClient"/> associated with this client.
```

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: Google.Api.Generator.Tests/ProtoTests/Showcase/Google.Showcase.V1Beta1/IdentityClient.g.cs (hunk 1 above)
Comment (jskeet) [reply Google.Api.Generator.Tests/ProtoTests/Showcase/Google.Showcase.V1Beta1/IdentityClient.g.cs:124]: This is purely generated - we can change ServiceSettingsCodeGenerator at another time, but it shouldn't be part of this PR.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Pull request merged**